### ISSUE TITLE
What is the definition of n_period in BetaGeoBetaBinomFitter.fit?

### ISSUE THREAD

**Issue by hmikelee** [CamDavidsonPilon/lifetimes#353]
Dear Sir,

Thank you for your great package, I have applied BG/NBD to different data sets in retail and banking industries with good results. Really appreciate the great work.

I am now testing to see if BG/BB can provide better results. While preparing the data, I am not sure if the field n_period should be the # of days of the training data set or n_period is the same as T in BetaGeoFitter.fit

I am not able to find any example online. Your help will be highly appreciated. Thank you very much

hmikelee

**Comment by psygo** [comment]
I'm not sure, but I think you should take a look at the original paper in order to find out more about it:

Fader, Peter S., Bruce G.S. Hardie, and Jen Shang (2010), *"Customer-Base Analysis in a Discrete-Time Noncontractual Setting,"*, Marketing Science, 29 (6), 1086-1108.

**Comment by hmikelee** [comment]
Dear Philippe,

Sorry that I did not express my questions more thoroughly.

1. The example used in the paper you mentioned above used an example of a 1995 cohort, ie. customers with the same starting time of 1995. I am just wondering that if we have a population with different starting time, can we put them in the same dataset for fit to estimate parameters or we will need to fit different cohorts with different starting time?
2. If we can put different cohorts into the same data set for fit to find parameters, and the n_period can be used to denote different start time cohorts. In this case, n_period is the same as T for other fitters in the lifetime package. I did run the same data sets with BG/NBD and BG/DD in lifetimes, the forecast results are drastically different 

I would guess that the data preparation for BG/NBD and BG/DD are different. That was the reason I ask the reason about n_period

Thank you very much for your support and patience
